Fix issues with `setFirstResult(0)` and MSSQL

DavideD opened the pull request:
Fixes #931 

Not an elegant solution but should fix the issues

gavinking commented on hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/SQLServerParameters.java:
```diff
@@ -34,24 +36,48 @@ public String process(String sql, int parameterCount) {
+            String sqlProcessed = "select top(@P" + index + ")" + sql.substring( 13 );
+            shiftValues( parameterArray );
+            return sqlProcessed;
+        }
```
It's unfortunate that ORM sometimes generates such ugly SQL. This limit handler stuff is such a mess in 5. I almost feel like back porting the cleanups I did for H6. :-/

DavideD replied:
It would be nice, but I would wait and see if other more complicated use cases pop up before refactoring 5.
Unless is something relatively easy to do.

gavinking commented on hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/SQLServerParameters.java:
```diff
@@ -34,24 +36,48 @@ public String process(String sql, int parameterCount) {
+        else {
+            pos = sql.indexOf( " offset 0 rows fetch next ? rows only" );
+            offsetQueryString = " offset 0";
+        }
```
@DavideD wouldn't it be good enough to just match `" offset "` and then grab the next two characters?

DavideD replied:
Nice idea.
It bothers me a bit that we check such a short string but I've updated the code anyway

gavinking commented:
@Davide, I'm inclined to say: simplify this patch + just not support SQL Server versions earlier than 2012.

Instead of maintaining that rather ugly, (and also currently untested) code for the `SQLServer2005LimitHandler`.

gavinking commented:
> Sounds good, I will update the pr in a bit

👍

DavideD commented:
Wait ... this issue doesn't seem related to the version of the limit handler. We are  using `SQLServer2012LimitHandler` and the test fails anyway. Maybe I didn't understand what you are suggesting?

gavinking commented:
I mean the specific bit of the fix which handles `select top(?)`. `SQLServer2012LimitHandler` doesn't use `top`.

DavideD commented:
> I mean the specific bit of the fix which handles `select top(?)`. `SQLServer2012LimitHandler` doesn't use `top`.

I've checked and `SQLServer2012LimitHandler` still uses `top` when the query doesn't have a order by clause.
I've rebased and updated the comment in my previous pr

DavideD commented:
I'm going to merge this one. We can revert it or fix bugs later
